### PR TITLE
THE WE ARE GOON 2.0 UPDATE

### PR DESCRIPTION
**Pull request by MacHac**
### Intent of Pull Request
- Ports civilian objectives from goonstation
- Ports cannabis from goonstation
- Adds new chemical THC (ported from goonstation, only occurs in cannabis)

Civilian objectives are objectives that nonantagonists can recieve.  These allow for some limited form of greentext, rewarding such great feats of strength as keeping your hat on as captain.  Approximately 10% of the crew will recieve such an objective every round, and their embarrassing failure will be put on display for all to see just like a real antag.  This has not been tested at all for the moment because you need at least five people on a server to have a chance of anyone getting a civilian objective.  The fact that this hasn't been tested means that **merging this is probably a bad idea until it gets more testing**.

Cannabis was ported partially out of a desire to make the botanist objective work and partially out of a desire to not have botanists getting high on Ambrosia Vulgaris.  Cannabis comes in five flavors: Vanilla, or regular; black, or poisonous; white, or healing; mega, or unusually strong; and omega, which needs no introduction.  They do work, but may need some spriting done.  Specifically, all of their seed packets currently look like Ambrosia Vulgaris packets because goon doesn't use custom seed packet icons.

My reason for putting this up now is so that you lovely people can flip through my code and poke holes in it, so that I can try to remove some of the bugs before they set the server on fire.  You are also welcome to suggest new objectives, or modifications to the existing ones.

### CODE REVIEW

**Comment by ShadowDeath6** [comment]
Medical Director would need to be CMO if the coding needs to correspond with actual jobs. Also you'd need to transfer the robotics objectives from CMO/Med Director to RD since robotics is only part of medical in goon.

**Comment by oisin100** [comment]
**WARNING OCD ALERT INCOMING**

It would be nice if you could convert the goonstation datastructure into tgstation structure...

File: code/game/gamemodes/civilian_objectives.dm [hunk 1]
```diff
@@ -0,0 +1,451 @@
...
+
+	headofsecurity
+		hat
+			explanation_text = "Have your beret on at the end of the round."
+			check_completion()
+				if(owner.current && owner.current.check_contents_for(/obj/item/clothing/head/beret/sec/navyhos))
+					return 1
+				else
+					return 0
+		brig
```
Comment (AsV9) [inline code/game/gamemodes/civilian_objectives.dm]: Cue HoS leaving 5 people bucklecuffed at brig when the shuttle leaves.

File: code/game/gamemodes/civilian_objectives.dm [hunk 2]
```diff
@@ -0,0 +1,451 @@
...
+			explanation_text = "Have at least one antagonist cuffed in the brig at the end of the round." //can be dead as people usually suicide
+			check_completion()
+				for(var/datum/mind/M in ticker.minds)
+					if(M.special_role && M.current && !istype(M.current,/mob/dead) && istype(get_area(M.current),/area/prison)) //think that's everything...
+						var/mob/living/carbon/C = M.current //Oh the things we do to avoid the colon
+						if(!C.handcuffed)
+							return 1
+				return 0
```
Comment (AsV9) [inline code/game/gamemodes/civilian_objectives.dm:36]: Cue HoS leaving 5 people bucklecuffed at brig when the shuttle leaves.

**Comment by DavidEdwards** [comment]
We need to get these files converted to standardized code style.

So:

`/datum/objective/crew/captain/hat`

Instead of:

```
/datum/objective/crew
    captain
        hat
```

It looks like Goon code has a different style standard.

File: code/game/gamemodes/civilian_objectives.dm [hunk 3]
```diff
@@ -0,0 +1,451 @@
...
+	chaplain
+		funeral
+			explanation_text = "Have no corpses on the station level at the end of the round."
+			check_completion()
+				for(var/mob/living/carbon/human/H in world)
+					if(H.z == 1 && H.stat == 2)
+						return 0
+				return 1
+
+	janitor
```
Comment (ToGWtF) [inline code/game/gamemodes/civilian_objectives.dm:181]: No mice alive objective would be fun

File: code/game/gamemodes/civilian_objectives.dm [hunk 4]
```diff
@@ -0,0 +1,451 @@
...
+					if(M.special_role && M.current && !istype(M.current,/mob/dead) && istype(get_area(M.current),/area/shuttle/escape))
+						if(owner.current && owner.current.stat != 2 && istype(get_area(owner.current),/area/shuttle/escape)) //split this up as it was long
+							var/mob/living/carbon/C = M.current
+							if(C.handcuffed)
+								return 1
+				return 0
+
+	headofpersonnel
+		vanish
+			explanation_text = "End the round alive but not on the station or escape levels."
```
Comment (Alek2ander) [inline code/game/gamemodes/civilian_objectives.dm:50]: This is literally against the server rules.

File: code/game/gamemodes/civilian_objectives.dm [hunk 5]
```diff
@@ -0,0 +1,451 @@
...
+	headofsecurity
+		hat
+			explanation_text = "Have your beret on at the end of the round."
+			check_completion()
+				if(owner.current && owner.current.check_contents_for(/obj/item/clothing/head/beret/sec/navyhos))
+					return 1
+				else
+					return 0
+		brig
+			explanation_text = "Have at least one antagonist cuffed in the brig at the end of the round." //can be dead as people usually suicide
```
Comment (Alek2ander) [inline code/game/gamemodes/civilian_objectives.dm:29]: This is pure shitcurity - will encourage the HoS to bucklecuff people in permabrig and never bring them to the shuttle

File: code/game/gamemodes/civilian_objectives.dm [hunk 6]
```diff
@@ -0,0 +1,451 @@
...
+						return 0
+				for (var/obj/machinery/hydroponics/PP in machines)
+					if (PP.myseed && (istype(PP.myseed, /obj/item/seeds/cannabis_seed) || istype(PP.myseed, /obj/item/seeds/mega_cannabis_seed) || istype(PP.myseed, /obj/item/seeds/white_cannabis_seed) || istype(PP.myseed, /obj/item/seeds/black_cannabis_seed) || istype(PP.myseed, /obj/item/seeds/omega_cannabis_seed)))
+						if (istype(get_area(PP), /area/hydroponics) || istype(get_area(PP), /area/hydroponics/backroom))
+							return 0
+				return 1
+
+	chaplain
+		funeral
+			explanation_text = "Have no corpses on the station level at the end of the round."
```
Comment (Alek2ander) [inline code/game/gamemodes/civilian_objectives.dm:174]: Well this one is impossible

File: code/game/gamemodes/civilian_objectives.dm [hunk 7]
```diff
@@ -0,0 +1,451 @@
...
+						for(var/fert in fert_types)
+							if(content.reagents.has_reagent(fert,1))
+								fert_count++
+								fert_types -= fert
+
+				if(fert_count >= 4)
+					return 1
+				return 0
+		noweed
+			explanation_text = "Make sure there are no cannabis plants, seeds or products in Hydroponics at the end of the round."
```
Comment (Alek2ander) [inline code/game/gamemodes/civilian_objectives.dm:130]: They'll just grow ambrosia, this is pointless

File: code/game/gamemodes/civilian_objectives.dm [hunk 8]
```diff
@@ -0,0 +1,451 @@
...
+				return 0
+		hyper
+			explanation_text = "Have methamphetamine in your bloodstream at the end of the round."
+			check_completion()
+				if(owner.current && owner.current.reagents && owner.current.reagents.has_reagent("methamphetamine"))
+					return 1
+				else
+					return 0
+		onfire
+			explanation_text = "Escape on the shuttle alive while on fire with silver sulfadiazine in your bloodstream."
```
Comment (Alek2ander) [inline code/game/gamemodes/civilian_objectives.dm:246]: "Griff the entire shuttle by running around setting everyone on fire as nonantag." Yeah, great idea.

File: code/game/gamemodes/civilian_objectives.dm [hunk 9]
```diff
@@ -0,0 +1,451 @@
...
+	scientist
+		hyper
+			explanation_text = "Have methamphetamine in your bloodstream at the end of the round."
+			check_completion()
+				if(owner.current && owner.current.reagents && owner.current.reagents.has_reagent("methamphetamine"))
+					return 1
+				else
+					return 0
+		onfire
+			explanation_text = "Escape on the shuttle alive while on fire with silver sulfadiazine in your bloodstream."
```
Comment (Alek2ander) [inline code/game/gamemodes/civilian_objectives.dm:262]: "Griff the entire shuttle by running around setting everyone on fire as nonantag." Yeah, great idea.

File: code/game/gamemodes/civilian_objectives.dm [hunk 10]
```diff
@@ -0,0 +1,451 @@
...
+			explanation_text = "Have your appendix removed somehow by the end of the round." //Give the doctors something to do
+			check_completion()
+				if(owner.current && ishuman(owner.current))
+					var/mob/living/carbon/human/H = owner.current
+					for(var/obj/item/organ/internal/O in H.internal_organs)
+						if(istype(O,/obj/item/organ/internal/appendix))
+							return 0
+				return 1
+		clown
+			explanation_text = "Escape on the shuttle alive wearing at least one piece of clown clothing."
```
Comment (Alek2ander) [inline code/game/gamemodes/civilian_objectives.dm:434]: Pls greytide more tnx

File: code/game/gamemodes/civilian_objectives.dm [hunk 11]
```diff
@@ -0,0 +1,451 @@
...
+			check_completion()
+				if(owner.current && ishuman(owner.current))
+					var/mob/living/carbon/human/H = owner.current
+					if(istype(H.wear_mask,/obj/item/clothing/mask/gas/clown_hat) || istype(H.w_uniform,/obj/item/clothing/under/rank/clown) || istype(H.shoes,/obj/item/clothing/shoes/clown_shoes)) return 1
+				return 0
+		spacesuit
+			explanation_text = "Get your grubby hands on a spacesuit."
```
Comment (Alek2ander) [inline code/game/gamemodes/civilian_objectives.dm:441]: Pls greytide more tnx

File: code/game/gamemodes/civilian_objectives.dm (hunk 4 above)
Comment (Iamgoofball) [inline code/game/gamemodes/civilian_objectives.dm:50]: why is this against server rules

File: code/game/gamemodes/civilian_objectives.dm (hunk 7 above)
Comment (Iamgoofball) [inline code/game/gamemodes/civilian_objectives.dm:130]: add ambrosia to the checklist

File: code/game/gamemodes/civilian_objectives.dm [hunk 12]
```diff
@@ -0,0 +1,451 @@
...
+					if(C.records.len > 4)
+						return 1
+				return 0
+		noscorch
+			explanation_text = "Ensure that the floors of the chemistry lab are not scorched at the end of the round."
+			check_completion()
+				for(var/turf/simulated/floor/T in get_area_turfs(/area/medical/chemistry, 0))
+					if(T.burnt == 1) return 0
+				return 1
+		/* Commented out because why would the MD control that
```
Comment (Iamgoofball) [inline code/game/gamemodes/civilian_objectives.dm:297]: For reference. Robotics is a medbay thing and only does cyborgs in gooncode

File: code/game/gamemodes/civilian_objectives.dm [hunk 13]
```diff
@@ -0,0 +1,451 @@
...
+						return 1
+				return 0
+		scanned
+			explanation_text = "Have at least 5 people's DNA scanned in the cloning console at the end of the round."
+			check_completion()
+				for(var/obj/machinery/computer/cloning/C in machines)
+					if(C.records.len > 4)
+						return 1
+				return 0
+		noscorch
```
Comment (Iamgoofball) [inline code/game/gamemodes/civilian_objectives.dm:291]: Also, chemistry is a science department on goon, so medical is less involved. Hence the RD's objectives with chemicals

File: code/game/gamemodes/civilian_objectives.dm (hunk 4 above)
Comment (ghost) [inline code/game/gamemodes/civilian_objectives.dm:50]: It's not against server rules, he's being an dumb

File: code/game/gamemodes/civilian_objectives.dm (hunk 5 above)
Comment (ghost) [inline code/game/gamemodes/civilian_objectives.dm:29]: So what, permabrigged people don't have to be brought to the shuttle, doesn't matter, they're traitors and can be left to rot in permabrig, including the one asshole ling you don't have time to cremate so you weld all the vents and shove them in iso with blast doors shut and doors welded

File: code/game/gamemodes/civilian_objectives.dm (hunk 6 above)
Comment (ghost) [inline code/game/gamemodes/civilian_objectives.dm:174]: Get good

File: code/game/gamemodes/civilian_objectives.dm (hunk 8 above)
Comment (ghost) [inline code/game/gamemodes/civilian_objectives.dm:246]: How about 'Escape on the shuttle with CLF3/Phlogistion and leporazine in your bloodstream

File: code/game/gamemodes/civilian_objectives.dm (hunk 10 above)
Comment (ghost) [inline code/game/gamemodes/civilian_objectives.dm:434]: shoo

File: code/game/gamemodes/civilian_objectives.dm (hunk 11 above)
Comment (ghost) [inline code/game/gamemodes/civilian_objectives.dm:441]: it's fine, shoo

File: code/game/gamemodes/civilian_objectives.dm (hunk 4 above)
Comment (Alek2ander) [inline code/game/gamemodes/civilian_objectives.dm:50]: 2.4.2: Never abandon the station. When playing a command role, do not leave the station z-level without a good reason.

File: code/game/gamemodes/civilian_objectives.dm (hunk 5 above)
Comment (Alek2ander) [inline code/game/gamemodes/civilian_objectives.dm:29]: **Cuffed** in the brig.

Comment (ghost) [inline code/game/gamemodes/civilian_objectives.dm:29]: Not an issue

Permabrigged prisoners have no rights and can be left bucklecuffed

File: code/game/gamemodes/civilian_objectives.dm (hunk 4 above)
Comment (ghost) [inline code/game/gamemodes/civilian_objectives.dm:50]: 'Ask cap to leave'

'yeah'

win

also

if he gets the obj it's not against server rules

Comment (Iamgoofball) [inline code/game/gamemodes/civilian_objectives.dm:50]: ^^^^^^^^^

**Comment by MacHac** [comment]
I've seen a lot of good criticisms so far, and i'm going to give the arguments another 24 hours or so to settle down.  Does anyone have any ideas for other objectives to replace the ones that will inevitably be removed?  Keep in mind that several professions currently don't have any objectives.

File: code/game/gamemodes/civilian_objectives.dm (hunk 6 above)
Comment (Notamaniac) [inline code/game/gamemodes/civilian_objectives.dm:174]: Only if the mass driver decides that it's a good day and doesn't get stuck launching. Or the space torus loops the corpse back to the otherside of the station

Comment (Cruix) [inline code/game/gamemodes/civilian_objectives.dm:174]: ";Hey, why are there 30 corpses outside of arrivals?"
*Chaplain starts breaking into EVA

**Comment by MacHac** [comment]
It occurs to me that I would have to reformat all of this code in order to get this PR accepted, so I'm closing this for now.

**Comment by oisin100** [comment]
Its still a nice thing to give the crew and i would still love to see if implemented. Its quite the job reformatting all the code but you dont need to worry about bugs and all that, Try giving it a go!

**Comment by MacHac** [comment]
I said for now, oisin.  I'll get around to it on some lazy day where there's nothing better to do than to reformat 500 lines of code according to the style guide.  Probably this weekend.  I just didn't want it clogging up the PR queue while I worked on it.

**Comment by ghost** [comment]
lad hey lad @MacHac

**Pull request closed without merging**